lib/orderbook/OrderBook.ts

```diff
@@ -946,17 +946,22 @@ class OrderBook extends EventEmitter {
   public removeOwnOrders = () => {
     const removedOrderLocalIds = [];
     const onHoldOrderLocalIds = [];
+    const erroredOrdersByLocalId = new Map<string, string>();
 
     for (const localId of this.localIdMap.keys()) {
-      const { onHoldQuantity } = this.removeOwnOrderByLocalId(localId, true);
-      if (onHoldQuantity === 0) {
-        removedOrderLocalIds.push(localId);
-      } else {
-        onHoldOrderLocalIds.push(localId);
+      try {
+        const { onHoldQuantity } = this.removeOwnOrderByLocalId(localId, true);
+        if (onHoldQuantity === 0) {
+          removedOrderLocalIds.push(localId);
+        } else {
+          onHoldOrderLocalIds.push(localId);
+        }
+      } catch (ex) {
+        erroredOrdersByLocalId.set(localId, ex.message);
```

Comment:
@sangaman what do you think of this approach? Shouldn't `removeallorders` work in any case or is there a genuine case when we cannot remove some specific orders?

Reply:
Since removeorder throws exception in any case of failures, maybe we can extend to catch some specific errors here, will wait @sangaman 's feedback

Reply:
Yes I don't think this is the right approach. I suspect there may be orders that aren't getting removed from `localIdMap` even when they are removed from the rest of the order book, therefore we may sometimes be removing orders that no longer exist. That's what throws an error and causes the issue.

I'm interested in figuring out *why* we get not found errors here in the first place, but that can be a separate PR. Fow now, I think all that needs to be done is to log a `debug` message if the error has the code for `ORDER_NOT_FOUND` saying that we tried to remove an order that no longer exists, and log an error message in any other case. There *shouldn't* be any cases where we fail to remove an order, and if the reason we can't remove it is because it doesn't exist, I don't think we need to return anything via rpc about that order since it is indeed removed and the end state will be that there are no orders remaining, which is what we want.